src/fcmcts/fcmcts.jl

```diff
@@ -259,31 +259,31 @@ function compute_best_action_node(mdp::JointMDP, tree::FCMCTSTree, node::FCState
     return best
 end
 
-POMDPLinter.@POMDP_require simulate(planner::FCMCTSPlanner, s, depth::Int64) begin
+@POMDP_require simulate(planner::FCMCTSPlanner, s, depth::Int64) begin
     mdp = planner.mdp
     P = typeof(mdp)
     @assert P <: JointMDP       # req does different thing?
-    SV = statetype(P)
-    @assert typeof(SV) <: AbstractVector # TODO: Is this correct?
+    #SV = statetype(P)
+    #@assert typeof(SV) <: AbstractVector # TODO: Is this correct?
     AV = actiontype(P)
-    @assert typeof(A) <: AbstractVector
+    @assert typeof(AV) <: AbstractVector
     @req discount(::P)
     @req isterminal(::P, ::SV)
     @subreq insert_node!(planner.tree, planner, s)
     @subreq estimate_value(planner.solved_estimate, mdp, s, depth)
-    @req gen(::DDNOut{(:sp, :r)}, ::P, ::SV, ::A, ::typeof(planner.rng))
+    @req gen(::DDNOut{(:sp, :r)}, ::P, ::SV, ::AV, ::typeof(planner.rng))
 
     # MMDP reqs
-    @req get_agent_actions(::P, ::Int64)
-    @req get_agent_actions(::P, ::Int64, ::eltype(SV))
+    @req agent_actions(::P, ::Int64)
+    @req agent_actions(::P, ::Int64, ::eltype(SV)) # TODO should this be eltype?
```

Comment:
I think this is fine as we do use `enumerate(s)` above. We are implicitly requiring that the datastructure be something we can apply `eltype` to

Reply:
Yeah, this is what I am unsure regarding @zsunberg's comments. We are assuming that `S` for JointMDP is something we can enumerate into agent specific things (which might just be copies of the same thing), but there could instead just be a single global state too.

Reply:
I think this is fine. Worst case it tells the problem writer that they need to implement `agent_actions(::P, ::Int64, ::Any)`